### PR TITLE
chore: Improve the Base64Provider resilience to malformed configuration files

### PR DESCRIPTION
**Pull request by 0xced**
Also add some tests to cover different pathological scenarios.

### CODE REVIEW

File: tests/Testcontainers.Tests/Unit/Builders/Base64ProviderTest.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+using System.Text.Json;
+using DotNet.Testcontainers.Builders;
+using Microsoft.Extensions.Logging.Abstractions;
+using Xunit;
+
+namespace DotNet.Testcontainers.Tests.Unit
+{
+  public class Base64ProviderTest
```
Comment (HofmeisterAn) [inline tests/Testcontainers.Tests/Unit/Builders/Base64ProviderTest.cs]: Can we  please move these tests to the already existing `Base64ProviderTest` test class? It is nested in `DockerRegistryAuthenticationProviderTest`. Probably we can reuse the existing test methods `ShouldGetAuthConfig` and `ShouldGetNull`.

File: src/Testcontainers/Builders/Base64Provider.cs [hunk 2]
```diff
@@ -88,5 +96,18 @@ public IDockerRegistryAuthenticationConfiguration GetAuthConfig(string hostname)
...
+    private static byte[] DecodeBase64String(string base64)
+    {
+      try
+      {
+        return Convert.FromBase64String(base64);
+      }
+      catch (FormatException)
+      {
+        return null;
+      }
```
Comment (HofmeisterAn) [inline src/Testcontainers/Builders/Base64Provider.cs]: Should we write a log message in case of an exception? It is likely a misconfiguration?

Comment (0xced) [reply src/Testcontainers/Builders/Base64Provider.cs]: Good point! I added logs for every code path where the `auth` part can't be decoded in f79827f2ba376b4e29be8327d08f89961252442f.

File: tests/Testcontainers.Tests/Unit/Builders/Base64ProviderTest.cs (hunk 1 above)
Comment (0xced) [reply tests/Testcontainers.Tests/Unit/Builders/Base64ProviderTest.cs]: Absolutely, I got confused by the fact the existing `Base64ProviderTest` does not follow the same directory structure as in `src`. I have moved all the tests into the existing class in acd009ecea51ea89c9cbd232006a6bd426cd730c.

**Review by HofmeisterAn — APPROVED** [review]
Thanks 🎉.

**Pull request merged**